[Ricky Andrew] iP

rickyaandrew opened the pull request:
### Duke Bot Todo App <hr>
> Serve to make your day more productive :) - Duke [(cite)](github.com) 

Duke will make sure that you will not miss any important matters in your life, making sure that you take down your todo list every single day. Duke is:

1. Easy to use
2. Lightweight
3. Simple in design

What you need to do is:

- Download this todo app
- PayLah me $100 at 98765432
- Add your task in the todo app
- Allow Duke to manage your tasks today

Feel free to donate more money via PayLah at 98765432.

Features:

- [x] This app is still developing
- [x] Simple and easy to use
- [x] Duke bakes bread and is nice
- [x] Duke has its own bread shop called Duke Bakery
- [x] Duke also manages your todo list for the day
- [ ] ~Duke is trying out this strike feature~
- [ ] **Duke is trying out this bold feature**
- [ ] _Duke is trying out the italic feature_
- [x] Duke will stay by your side forever unlike anyone else ❤️ 

Please do look at my `main` method here. Its simple and easy to understand. Can finish reading it within seconds!
```
    /**
     * Runs the main Duke ToDo Application
     * 
     * @param args
     * @throws FileNotFoundException when file is not found in the given location.
     * @throws IOException           when Scanner is not able to scan anything from
     *                               the given file path.
     */
    public static void main(String[] args) throws FileNotFoundException, IOException {
        Duke duke = new Duke();
        duke.start();
    }
```

Commit:
Add Gradle support

rickyaandrew committed:
Level-1

rickyaandrew committed:
Level-1

rickyaandrew committed:
Level-1

rickyaandrew committed:
Level-2 commits

rickyaandrew committed:
Level-2 clean up

rickyaandrew committed:
Level-3

rickyaandrew committed:
Level-4 done

rickyaandrew committed:
Level-4 cleanup

rickyaandrew committed:
fixed bugs

rickyaandrew committed:
A-TextUiTesting

rickyaandrew committed:
Level-5

rickyaandrew committed:
Level-6

rickyaandrew committed:
small enhancements

rickyaandrew committed:
Merge pull request #1 from rickyaandrew/ricky

PR for Week 2 Individual Project

rickyaandrew committed:
Add data.txt Done Level-7 Save

rickyaandrew committed:
Add DateTime formatting for Deadline

rickyaandrew committed:
Resolve conflicts

rickyaandrew committed:
Add Ui and Storage

rickyaandrew committed:
Add TaskList and Update Ui

rickyaandrew committed:
Fix formatting bugs

rickyaandrew committed:
Package and Organise files

rickyaandrew committed:
Add JUnit Test for Deadline, Events and ToDos

rickyaandrew committed:
Add TaskListTest for JUnit Test

rickyaandrew committed:
Add JavaDocs into all the .java files

rickyaandrew committed:
Add features to follow coding standard

rickyaandrew committed:
completed branch-A-CodingStandard

rickyaandrew committed:
Merge with A-JavaDoc

rickyaandrew committed:
Add find feature

rickyaandrew committed:
Add JavaDocs for new functions

rickyaandrew committed:
Merge with branch-A-JavaDoc

rickyaandrew committed:
Edit gitignore

rickyaandrew committed:
Edit gitignore

rickyaandrew committed:
Merge branch 'add-gradle-support' of https://github.com/rickyaandrew/ip into add-gradle-support

rickyaandrew committed:
Add Exceptions with proper formatting

rickyaandrew committed:
Edit on missing line

rickyaandrew committed:
Merge branch 'master' of https://github.com/rickyaandrew/ip into add-gradle-support

rickyaandrew committed:
Bug fixes

rickyaandrew committed:
Fix filepath issues

rickyaandrew committed:
Edit on tests

rickyaandrew committed:
Gradle is working

rickyaandrew committed:
Start on launching an Application

xlzior commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+    private final Storage storage;
+
+    /**
+     * Class Constructor.
```
Perhaps this JavaDoc should be changed to follow convention?

From the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#comments):
> In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returning etc.)

Also in line with Java's built-in class conventions: https://github.com/nus-cs2103-AY2122S1/forum/issues/82

xlzior commented on src/main/java/seedu/data/data.txt:
```diff
@@ -0,0 +1,5 @@
+E | 1 | project meeting | Aug 6th 2-4pm
```
Perhaps this should be added to `.gitignore`?

xlzior commented on src/main/java/seedu/duke/commands/Ui.java:
```diff
@@ -0,0 +1,74 @@
+    public static final String ERROR_MESSAGE_STORAGE_SAVE = "DukeStorageSaveException: Unable to write data into storage.";
+    public static final String ERROR_MESSAGE_STORAGE_LOAD = "DukeStorageLoadException: Unable to locate and load the data from storage.";
+    public static final String ERROR_MESSAGE_STORAGE_LOAD_OUT_OF_BOUND = "DukeStorageLoadException: Data must have saved in a wrong format.";
+    public static final String ERROR_MESSAGE_ACTION_OUT_OF_BOUND = "DukeActionOutOfBoundException: There is no such index from the storage.";
```
I like your use of constants in this manner, it makes things very clear and easy to modify!

xlzior commented on src/main/java/seedu/duke/exceptions/storage/DukeStorageUpdateException.java:
```diff
@@ -0,0 +1,10 @@
+package seedu.duke.exceptions.storage;
```
I like your detailed and nested structure for packages, especially exceptions.

xlzior commented on src/main/java/seedu/duke/storage/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+    }
+
+    /**
+     * Helps to mark the task in the list of Tasks as done.
```
Is there a reason why some of your JavaDocs comments use `Helps..`? Would it be easier/clearer to just say `Marks...` etc.?

xlzior commented on src/main/java/seedu/data/data.txt:
```diff
@@ -0,0 +1,5 @@
+E | 1 | project meeting | Aug 6th 2-4pm
```
Should this file be added to `.gitignore`?

clyveycui replied:
Second this.

xlzior commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+    private final Storage storage;
+
+    /**
+     * Class Constructor.
```
Perhaps this JavaDoc should be changed to follow convention?

From the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html#comments):
> In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returning etc.)

Also in line with Java's built-in class conventions: https://github.com/nus-cs2103-AY2122S1/forum/issues/82

clyveycui replied:
Javadoc for constructors aren't necessary btw.

xlzior commented on src/main/java/seedu/duke/commands/Ui.java:
```diff
@@ -0,0 +1,74 @@
+    public static final String ERROR_MESSAGE_STORAGE_SAVE = "DukeStorageSaveException: Unable to write data into storage.";
+    public static final String ERROR_MESSAGE_STORAGE_LOAD = "DukeStorageLoadException: Unable to locate and load the data from storage.";
+    public static final String ERROR_MESSAGE_STORAGE_LOAD_OUT_OF_BOUND = "DukeStorageLoadException: Data must have saved in a wrong format.";
+    public static final String ERROR_MESSAGE_ACTION_OUT_OF_BOUND = "DukeActionOutOfBoundException: There is no such index from the storage.";
```
I like your use of constants in this manner, it makes things very clear and easy to modify.

SpdPnd98 replied:
Second this!

xlzior commented on src/main/java/seedu/duke/exceptions/storage/DukeStorageDeleteException.java:
```diff
@@ -0,0 +1,10 @@
+package seedu.duke.exceptions.storage;
```
I like your detailed and nested structure for packages, especially exceptions.

xlzior commented on src/main/java/seedu/duke/storage/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+    }
+
+    /**
+     * Helps to mark the task in the list of Tasks as done.
```
Is there a reason why some of your JavaDocs comments use `Helps...`? Would it be easier/cleaner to just say `Marks...` etc.?

Amadeus-Winarto replied:
Second this

xlzior reviewed (commented):
LGTM!

rickyaandrew committed:
Complete FXML tutorial guide

clyveycui commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+import java.util.ArrayList;
+
+public class Duke {
+    private final TaskList taskList;
+    private final Storage storage;
```
I think the variable names here should be in uppercase.

clyveycui commented on src/main/java/seedu/duke/commands/Command.java:
```diff
@@ -0,0 +1,21 @@
+     * 
+     * @return boolean whether the user wants to exit from the application.
+     */
+    public abstract boolean getIsExit();
```
I understand that it is a getter method, but I think it might be better to name the method `isExit` instead.

SpdPnd98 replied:
Second this. Especially due to the JavaDocs description. Based on implementations on other classes it doesn't seem to be a getter method rather a check method.

clyveycui reviewed (commented):
Lgtm! Minor fixes to be done, nothing major.

rickyaandrew committed:
Duke able to function in JavaFX

Amadeus-Winarto commented on src/main/java/seedu/duke/storage/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+            this.taskList.set(index, curr.markAsDone());
+            curr = this.taskList.get(index);
+
+            Ui.printMessage(Ui.DONE_MESSAGE + "\n" + Ui.INDENT + "  " + curr.toString());
```
In terms of separation of concerns, would it be better to place this Ui related part of the program outside of this class, which handles storage-specific tasks?

Amadeus-Winarto commented on src/main/java/seedu/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,60 @@
+     * @return a new Deadline object with the same description and by, but setting
+     *         isDone property to be true
+     */
+    @Override
+    public Deadline markAsDone() {
+        return new Deadline(this.getDescription(), this.getDateTime(), true);
+    }
```
Can I ask why not mutate the current Event?

SpdPnd98 replied:
I think mutating it would be a better design!

Amadeus-Winarto reviewed (commented):
Great job!

SpdPnd98 commented on src/main/java/seedu/data/data.txt:
```diff
@@ -0,0 +1,6 @@
+T | 1 | borrow book 3
+T | 1 | borrow hello 4
+T | 1 | borrow book 5
+T | 1 | borrow hello 6
+T | 1 | borrow book 7
+T | 1 | borrow hello 8
```
Should this file be added to gitignore?

SpdPnd98 commented on src/main/java/seedu/duke/commands/EmptyCommand.java:
```diff
@@ -0,0 +1,27 @@
+     * @return boolean whether the user wants to exit from the application.
+     */
+    @Override
+    public boolean getIsExit() {
+        return false;
+    }
+}
```
Similar comments with parent class. Also, overidden classes do not necessarily need javadocs, especially since it is so similar with the parent's docs.

SpdPnd98 commented on src/main/java/seedu/duke/commands/Ui.java:
```diff
@@ -0,0 +1,79 @@
+public class Ui {
+    public static final String logo = " ____        _        \n" + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n" + "| |_| | |_| |   <  __/\n" + "|____/ \\__,_|_|\\_\\___|\n";
+    // public static final String INDENTED_HORIZONTAL_LINE = "
+    // ____________________________________________________________";
+    // public static final String INDENT = " ";
+    // public static final String INTRODUCTION = "Hello! I'm Ricky\n" + Ui.INDENT +
+    // "What can I do for you?";
```
Don't leave commented code in the final version!

SpdPnd98 commented on src/main/java/seedu/duke/commands/Ui.java:
```diff
@@ -0,0 +1,79 @@
+        if (taskList.size() == 0) {
+            return Ui.printMessage(zeroSizeMsg);
+        }
+        // message = Ui.INDENTED_HORIZONTAL_LINE + "\n" + Ui.INDENT + listMsg + "\n";
+        message = listMsg + "\n";
+        for (int i = 0; i < taskList.size(); i++) {
+            // message += Ui.INDENT + (i + 1) + ". " + taskList.get(i).toString() + "\n";
+            message += (i + 1) + ". " + taskList.get(i).toString() + "\n";
+        }
+        // message += Ui.INDENTED_HORIZONTAL_LINE + "\n\n";
```
Don't leave commented code in final version.

SpdPnd98 commented on src/main/java/seedu/duke/tasks/Task.java:
```diff
@@ -0,0 +1,124 @@
+package seedu.duke.tasks;
+
+public class Task {
+    protected final String description;
+    protected final String dateTime;
+    protected final boolean isDone;
```
If you decide to make a Task mutable, remove `final` from these variables. Also, I think these should be `private` to the `Task` class, and shouldn't be accessible within the class (perhaps have a getter class to return these values).

SpdPnd98 reviewed (commented):
LGTM! Code is well structured, minor nit on certain code structure and naming conventions that others have mentioned.

rickyaandrew committed:
Review PR comments and made changes to it respectively

rickyaandrew committed:
Add Assertions on private functions to verify my assumptions on the argument passed.

There are several methods in Storage which have added assertions, so that when error occured, we will know exactly where when wrong.

Change some variable names as well to improve readability.

Easier for us to debug in the future as well, and allow users to understand the code easily.

rickyaandrew committed:
Review on code quality. Revised which includes Refactoring as well.

Includes refactoring so that the code is neater and cleaner for user to read and navigate. Applied Extractions to make the functions more readable.

Remove unwanted commented codes as well.

Such that the code will not be full of comments, as reader will read more meaningful codes and comments as well.

rickyaandrew committed:
Merge pull request #2 from rickyaandrew/branch-A-Assertions

Add Assertions on private functions to verify my assumptions on the a…

rickyaandrew committed:
Merge pull request #3 from rickyaandrew/branch-A-CodeQuality

Review on code quality. Revised which includes Refactoring as well.

rickyaandrew committed:
Add A-Streams into my code

rickyaandrew committed:
Merge branch 'master' of https://github.com/rickyaandrew/ip

rickyaandrew committed:
Fix delete bug and added some code on the GUI for Duke

rickyaandrew committed:
Add Enums to make the code neater

rickyaandrew committed:
Add some changes in UI and application icons

Add TimedTask

rickyaandrew committed:
Add PeriodTask

rickyaandrew committed:
Add DoAfterTasks

rickyaandrew committed:
Add Timetable and ScheduledTask yet to try

rickyaandrew committed:
Add scheduling deleting and accomodate schedule clashes

rickyaandrew committed:
Fix loading data from storage for TimedTask, ScheduledTask and PeriodTask

rickyaandrew committed:
Fixed ChainedTask issue, to load when reopened

rickyaandrew committed:
ViewSchedule is now sorted according to date and then time

rickyaandrew committed:
Minor bug fix

rickyaandrew committed:
Merge pull request #5 from rickyaandrew/redesign-code

Redesign code

rickyaandrew committed:
Adds JavaDocs. Have yet to clean up my code.

rickyaandrew committed:
SLAP my codes

rickyaandrew committed:
Merge pull request #6 from rickyaandrew/code-cleanup

i slap my codes

rickyaandrew committed:
Add doc-README.md

rickyaandrew committed:
Improves GUI

rickyaandrew committed:
Merge pull request #7 from rickyaandrew/better-GUI

Improves GUI

rickyaandrew committed:
Clean up

rickyaandrew committed:
Clears data

rickyaandrew committed:
Adds representative picture for Ui

rickyaandrew committed:
Updates README file

rickyaandrew committed:
Small Changes to fix upload for Fat Jar

rickyaandrew committed:
Added tagging 1

rickyaandrew committed:
Adds tagging functionality

rickyaandrew committed:
Merge pull request #8 from rickyaandrew/final-addons

Adds Tagging functionality

rickyaandrew committed:
Added Help command

rickyaandrew committed:
Merge pull request #9 from rickyaandrew/final-addons

Added Help command

rickyaandrew committed:
Date formatting

rickyaandrew committed:
Update README.md

rickyaandrew committed:
Minor fixes

rickyaandrew committed:
Formatting and Documentation

rickyaandrew committed:
Merge pull request #10 from rickyaandrew/final-addons

Bug fixes and JavaDocs

rickyaandrew committed:
README.md update
